bolt-sidecar/src/state/execution.rs

```diff
@@ -505,16 +505,16 @@ impl<C: StateFetcher> ExecutionState<C> {
     /// transactions by checking the nonce and balance of the account after applying the state
     /// diffs.
     fn refresh_templates(&mut self) {
-        for (address, (account_state, _)) in self.account_states.iter_mut() {
+        for (address, mut account_state) in self.account_states.iter().map(|(a, (s, _))| (*a, *s)) {
```

Comment:
Note: `copied` doesn't make it here because the return type of `iter` is `(&T, &(R, S))` instead of `&T`.

Reply:
Is there a way to make the behaviour a bit more clear when glancing through? 
It's not _super_ clear that we are mutating the copied version of the state. 

just my 2c, otherwise this is a great catch